tests/test_x509.py

```diff
@@ -1677,6 +1679,14 @@ def test_serial_number_may_only_be_set_once(self):
         with pytest.raises(ValueError):
             builder.serial_number(20)
 
+    def test_aware_not_valid_after(self):
+        time = datetime.datetime(2012, 1, 16, 22, 43)
+        tz = pytz.timezone("US/Pacific")
+        time = tz.localize(time)
+        utc_time = datetime.datetime(2012, 1, 17, 6, 43)
+        cert_builder = x509.CertificateBuilder().not_valid_after(time)
+        assert cert_builder._not_valid_after == utc_time
```

Comment:
This should generate a certificate and assert on that, not the internal property